### PR TITLE
chore : fix not null violation issue when creating fake chats for showing it on chat's index page

### PR DESCRIPTION
**Pull request by JaiveerChavda**
This PR fixes , integrity constraint violation: 19 NOT NULL chats.room_id issue.
actually when uncommenting chat factory line, creates this issue as it's not creating room record and try to insert hardcode id which doesn't exists. 

merging this PR will provide quicker development on chat module

image

### CODE REVIEW

File: app/Livewire/Chats/Index.php [hunk 1]
```diff
@@ -28,8 +28,10 @@ public function room(): ?Room
     public function render(): View
     {
         // add 10 chats to the database with the factory
-        // Chat::factory()->count(10)->create(['room_id' => $this->roomId]); // only uncomment once
-        // $this->roomId = 1;
+
+        // Chat::factory()->count(10)->create(['room_id' => Room::factory()->create()->id]); // only uncomment once
+        $this->roomId = 1;
```
Comment (MrPunyapal) [inline app/Livewire/Chats/Index.php:33]: Can you also add the same id here? 🤔

**Review by MrPunyapal — CHANGES_REQUESTED** [review]

File: app/Livewire/Chats/Index.php (hunk 1 above)
Comment (JaiveerChavda) [reply app/Livewire/Chats/Index.php:33]: sure , i wiil and sorry for late response

Comment (JaiveerChavda) [reply app/Livewire/Chats/Index.php:33]: BTW , what you're thoughts on streamlining this process of making room , chat and member in database seeder

Comment (MrPunyapal) [reply app/Livewire/Chats/Index.php:33]: We will remove all these comments and add seeders in the end.

These or all are for current development purpose 🫡

Comment (JaiveerChavda) [reply app/Livewire/Chats/Index.php:33]: okay got it.

File: app/Livewire/Chats/Index.php [hunk 2]
```diff
@@ -27,10 +28,17 @@ public function room(): ?Room
 
     public function render(): View
     {
+        // get the current auth user
+        $user = auth()->user(); 
+
+        // stop phpstan related errors by assert $user as instance of User model 
+        assert($user instanceof User);
```
Comment (MrPunyapal) [inline app/Livewire/Chats/Index.php:35]: we don't user assert()

File: app/Models/Room.php [hunk 3]
```diff
@@ -28,6 +28,16 @@ class Room extends Model
     /** @use HasFactory<RoomFactory> */
     use HasFactory;
 
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var list<string>
+     */
+    protected $fillable = [
```
Comment (MrPunyapal) [inline app/Models/Room.php:36]: we don't user fillable. (we will add unguarded in service provider later)

**Comment by MrPunyapal** [comment]
let skip this for now.

**Comment by JaiveerChavda** [comment]
okay, then how can we fix this current user issue.

**Comment by JaiveerChavda** [comment]
got it.

**Pull request closed without merging**